packages/agoric-cli/src/commands/psm.js

```diff
@@ -0,0 +1,136 @@
+    const { current: governance } = last(
+      storageNode.unserialize(govContent, fromBoard),
+    );
+    const { 'psm.IST.AUSD': instance } = agoricNames.instance;
+
+    return { instance, governance };
+  };
+
+  const showPublishedChildren = async () => {
+    // const status = await getJSON(`${RPC_BASE}/status?`);
+    // console.log({ status });
+    const raw = await getJSON(vstorage.url());
+    const top = vstorage.decode(raw);
+    console.error(
+      JSON.stringify(['vstorage published.*', JSON.parse(top).children]),
+    );
+  };
+
+  const showContractId = async showFees => {
+    const { instance, governance } = await getContractState();
+    showFees && console.error('psm', instance, Object.keys(governance));
+    showFees &&
+      console.error(
+        'WantMintedFee',
+        asPercent(governance.WantMintedFee.value),
+        '%',
+        'GiveMintedFee',
+        asPercent(governance.GiveMintedFee.value),
+        '%',
+      );
+    console.info(instance.boardId);
+  };
+
+  const showWallet = async addr => {
+    const state = await getWalletState(addr, fromBoard, {
+      getJSON,
+    });
+    const { assets, balances } = state;
+    // console.log(JSON.stringify(offers, null, 2));
+    // console.log(JSON.stringify({ offers, purses }, bigIntReplacer, 2));
+    const summary = {
+      balances: simplePurseBalances([...balances.values()], assets),
+      offers: simpleOffers(state, agoricNames),
+    };
+    console.log(fmtRecordOfLines(summary));
+    return 0;
+  };
+
+  const findOffer = async (addr, id) => {
+    const { assets, offers } = await getWalletState(addr, fromBoard, {
+      getJSON,
+    });
+    const offer = offers.get(id);
+    if (!offer) {
+      return 1;
+    }
+    const { numWantsSatisfied, payouts } = offer;
+    const fmt = makeAmountFormatter(assets);
+    const fmtRecord = r =>
+      Object.fromEntries(
+        Object.entries(r).map(([kw, amount]) => [kw, fmt(amount)]),
+      );
+    console.error({
+      numWantsSatisfied,
+      ...(payouts ? { payouts: fmtRecord(payouts) } : {}),
+    });
+    return typeof numWantsSatisfied === 'number' ? 0 : 1;
+  };
+
+  const marshaller = miniMarshal();
+
+  psm
+    .command('prepare')
+    .description('prepare an offer to the PSM')
+    .option('--wantMinted [DOLLARS')
+    .option('--wantAnchor [DOLLARS')
+    .option('--feePct [%]', 'Gas fee percentage')
+    .option('--offerId [number]', 'Offer id', String(Date.now()))
+    .option('--instance [id]', 'Instance id', 'psm-IST-AUSD')
+    .action(function () {
+      const opts = this.opts();
+      const { instance: instanceId } = opts;
+      const instance = agoricNames.instance[instanceId];
+      assert(instance, `name ${instanceId} not found`);
+      const spendAction = makePSMSpendAction(instance, agoricNames.brand, opts);
+      const capData = marshaller.serialize(spendAction);
+      console.log(JSON.stringify(capData));
```

Comment:
for future reference... while using console.log to print to stdout is idiomatic node.js, it's kinda cheating to rely on the console logger to have an observable effect. Ideally, we'd pass in `process.stdout` and write to that.